env0/resource_environment.go

```diff
@@ -241,10 +241,10 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
 			d.Set("template_id", environment.LatestDeploymentLog.BlueprintId)
 			d.Set("revision", environment.LatestDeploymentLog.BlueprintRevision)
 		}
-	} else if environment.BlueprintId != "" {
+	} else if environment.LatestDeploymentLog.BlueprintId != "" {
```

Comment:
@eladmosh - I think we should do an "or" just to be on the safe side.
If either is set use it...

Reply:
[checking for environment.BlueprintId also](https://github.com/env0/terraform-provider-env0/pull/509/commits/e81aa9b7b0e610d44e1e4ae528b1b8867c53eaab)